### PR TITLE
tests: prep lang category for the vm backend

### PR DESCRIPTION
**Pull request by saem**
## Summary
- ready to add the vm backend as lang categories default target
- lang category now runs for all categories

Changes:
- default the lang category to run on all relevant backends
- fixed test target declarations
- disabled some tests for broken backends like cpp

---

### CODE REVIEW

**Comment by saem** [comment]
@zerbina PTAL, this should help prep this category for the vm. I'm updating a few categories at a time, make it a bit easier for @canelhasmateus. :D

**Review by zerbina — APPROVED** [review]
Looks good to me, thanks! The ability to just include the VM target for relevant categories in ``defaultTargets`` will make things much easier.

File: tests/lang/s02_core/s02_procedures/t02_argument_passing.nim [hunk 1]
```diff
@@ -2,9 +2,10 @@ discard """
 description: '''
 Covers the multitude of ways you can pass arguments to procedures.
 '''
+targets: "c !cpp js"
```
Comment (zerbina) [inline tests/lang/s02_core/s02_procedures/t02_argument_passing.nim]: Would `targets: "!cpp"` also work?

Comment (saem) [reply tests/lang/s02_core/s02_procedures/t02_argument_passing.nim]: Ah, good catch, most likely

Comment (saem) [reply tests/lang/s02_core/s02_procedures/t02_argument_passing.nim]: fixed.

**Pull request merged**